src/main/scala/org/constellation/invertedmap/InvertedMap.scala

```diff
@@ -22,7 +22,6 @@ final class InvertedMap[A, B] private (m: Map[B, Set[A]])
     new InvertedMap(Inverted.update(m.toMap[B1, Set[A]], key, value))
 
   override def +[B1 >: B](kv: (A, B1)): InvertedMap[A, B1] = {
-    assert(!Inverted.contains(m, kv._1), "Given key already exists!")
```

Comment:
why? it's an important assertion and such a state is forbidden

Reply:
Regular Maps have `+` aliased to `updated`. Current API is counterintuitive, Map can be considered as a set of key-value pairs and adding to a set should be idempotent.

Reply:
as I wrote below, in this type of Map it's a forbidden state. it has been created purely for optimization and your current change breaks the important assumption about an Inverted Map.

Reply:
I'm marking this one as resolved to continue discussion in a single thread ([link](https://github.com/Constellation-Labs/constellation/pull/1474#discussion_r564935519)).